data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/EnumDeserializer.java

```diff
@@ -0,0 +1,57 @@
+package org.opensearch.dataprepper.pipeline.parser;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.BeanProperty;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.ContextualDeserializer;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+
+/**
+ * This deserializer is used for any Enum classes when converting the pipeline configuration file into the plugin model classes
+ * @since 2.11
+ */
+public class EnumDeserializer extends JsonDeserializer<Enum<?>> implements ContextualDeserializer {
+
+    static final String INVALID_ENUM_VALUE_ERROR_FORMAT = "Invalid value \"%s\". Valid options include %s.";
+
+    private Class<?> enumClass;
+
+    public EnumDeserializer() {}
+
+    public EnumDeserializer(final Class<?> enumClass) {
+        if (!enumClass.isEnum()) {
+            throw new IllegalArgumentException("The provided class is not an enum: " + enumClass.getName());
+        }
+
+        this.enumClass = enumClass;
+    }
+    @Override
+    public Enum<?> deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
+        final JsonNode node = p.getCodec().readTree(p);
+        final String enumValue = node.asText();
+
+        for (Object enumConstant : enumClass.getEnumConstants()) {
+            if (enumConstant.toString().equalsIgnoreCase(enumValue)) {
+                return (Enum<?>) enumConstant;
+            }
+        }
+
+        throw new IllegalArgumentException(String.format(INVALID_ENUM_VALUE_ERROR_FORMAT, enumValue,
+                Arrays.stream(enumClass.getEnumConstants()).map(valueEnum -> valueEnum.toString().toLowerCase()).collect(Collectors.toList())));
```

Comment:
Again, let's not assume these are always the same. Use `@JsonValue` instead.

Reply:
Can we assume that enums will always have a method annotated with `@JsonValue`, or should we fall back to default lowercase behavior if no method is found?

Reply:
No, we shouldn't assume this. Some may be missing. I think falling back is a good solution when there is no `@JsonValue`.